### PR TITLE
update calico to use canal

### LINKED ISSUE #1010: POD-to-POD networking broken when using networkPolicy Calico
**Is this an ISSUE or FEATURE REQUEST?** (choose one):

Issue

---

**What version of acs-engine?**:

v0.3.0

---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**

Kubernetes 1.6.x

**What happened**:

When deploying using networkPolicy set to Calico Pod-to-Pod networking is broken.

**What you expected to happen**:

Pods are able to connect with one another

**How to reproduce it** (as minimally and precisely as possible):

Deploy the Calico example API model

**Anything else we need to know**:

I've performed some troubleshooting and think the following is the best course of action currently. 

Calico is being deployed using the IP-in-IP as the transport along with NAT-outgoing being set. External (off-cluster) traffic is working file but Pod-to-Pod networking is broken due to IP-in-IP not being supported on Azure networking.

### Long-term fix
Have Azure CNI be the transport via CNI.

### Short-term fix
Given that we need a routing transport for Pod-to-Pod networking and Azure CNI is not finalized I propose we use CANAL which will utilize Flannel for transport.

### PR DESCRIPTION
**Pull request by lachie83**
Fixes #1010

---
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/1013)

### CODE REVIEW

**Comment by lachie83** [comment]
This needs documentation.

File: parts/kubernetesmasteraddons-calico-daemonset.yaml [hunk 1]
```diff
@@ -122,3 +150,20 @@ spec:
...
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: canal
+  namespace: kube-system
+  labels:
+    kubernetes.io/cluster-service: "true"
+    addonmanager.kubernetes.io/mode: "Reconcile"
```
Comment (seanknox) [inline parts/kubernetesmasteraddons-calico-daemonset.yaml]: Minor nit: add a ending line break

**Comment by seanknox** [comment]
@acs-bot test this please

**Comment by dmccaffery** [comment]
@seanknox should get some time to test this tonight; thanks!

**Comment by seanknox** [comment]
@lachie83 what documentation do we need to consider this complete?

**Comment by lachie83** [comment]
Just the details of the implementation. Probably this -- https://github.com/Azure/acs-engine/blob/master/docs/kubernetes.md#optional-enable-network-policy-enforcement-using-calico

**Comment by seanknox** [comment]
Let's get a blurb of documentation merge this.

**Comment by dmccaffery** [comment]
@lachie83 @seanknox as an FYI, this does seem to be working -- I was able to stage an environment with a custom vnet and pod->pod communications... I am still looking forward to CNI though. The ACI kubelet proxy seems intriguing as well, so I'll keep my eye on that.

Thanks for all the help!

File: parts/kubernetesmasteraddons-calico-daemonset.yaml (hunk 1 above)
Comment (lachie83) [inline parts/kubernetesmasteraddons-calico-daemonset.yaml]: Fixed

**Comment by lachie83** [comment]
@seanknox added docs. Please review

File: docs/kubernetes.md [hunk 2]
```diff
@@ -74,6 +74,8 @@ Using the default configuration, Kubernetes allows communication between all
 Pods within a cluster. To ensure that Pods can only be accessed by authorized
 Pods, a policy enforcement is needed. To enable policy enforcement using Calico refer to the [cluster definition](https://github.com/Azure/acs-engine/blob/master/docs/clusterdefinition.md#kubernetesconfig) document under networkPolicy. There is also a reference cluster definition available [here](https://github.com/Azure/acs-engine/blob/master/examples/networkpolicy/kubernetes-calico.json).
 
+The current implementation of Calico uses Flanal as the network overlay via CNI. The combination of Calico with Flanal is known as [Canal](https://github.com/projectcalico/canal/). Long-term, Flanal will be replaced with [Azure CNI](https://github.com/Azure/azure-container-networking/blob/master/docs/cni.md) as the provider for networking in this configuration.
```
Comment (jackfrancis) [inline docs/kubernetes.md]: I think "Flanal" should be "Flannel"

https://github.com/coreos/flannel

Comment (lachie83) [inline docs/kubernetes.md]: yep. It's one "n" in Aussie English but I'll change it for you.

**Comment by lachie83** [comment]
Investigating a strange error when RBAC isn't enabled.

```
Error from server (Forbidden): error when creating "calico-daemonset.yaml": clusterroles.rbac.authorization.k8s.io "canal" is forbidden: attempt to grant extra privileges: [PolicyRule{Resources:["namespaces"], APIGroups:[""], Verbs:["get"]} PolicyRule{Resources:["namespaces"], APIGroups:[""], Verbs:["list"]} PolicyRule{Resources:["namespaces"], APIGroups:[""], Verbs:["watch"]} PolicyRule{Resources:["pods"], APIGroups:[""], Verbs:["get"]} PolicyRule{Resources:["pods"], APIGroups:[""], Verbs:["list"]} PolicyRule{Resources:["pods"], APIGroups:[""], Verbs:["watch"]} PolicyRule{Resources:["nodes"], APIGroups:[""], Verbs:["get"]} PolicyRule{Resources:["nodes"], APIGroups:[""], Verbs:["list"]} PolicyRule{Resources:["nodes"], APIGroups:[""], Verbs:["watch"]} PolicyRule{Resources:["pods/status"], APIGroups:[""], Verbs:["update"]} PolicyRule{Resources:["nodes"], APIGroups:[""], Verbs:["update"]} PolicyRule{Resources:["thirdpartyresources"], APIGroups:["extensions"], Verbs:["create"]} PolicyRule{Resources:["thirdpartyresources"], APIGroups:["extensions"], Verbs:["get"]} PolicyRule{Resources:["thirdpartyresources"], APIGroups:["extensions"], Verbs:["list"]} PolicyRule{Resources:["thirdpartyresources"], APIGroups:["extensions"], Verbs:["watch"]} PolicyRule{Resources:["networkpolicies"], APIGroups:["extensions"], Verbs:["get"]} PolicyRule{Resources:["networkpolicies"], APIGroups:["extensions"], Verbs:["list"]} PolicyRule{Resources:["networkpolicies"], APIGroups:["extensions"], Verbs:["watch"]} PolicyRule{Resources:["globalbgppeers"], APIGroups:["projectcalico.org"], Verbs:["get"]} PolicyRule{Resources:["globalbgppeers"], APIGroups:["projectcalico.org"], Verbs:["list"]} PolicyRule{Resources:["globalconfigs"], APIGroups:["projectcalico.org"], Verbs:["create"]} PolicyRule{Resources:["globalconfigs"], APIGroups:["projectcalico.org"], Verbs:["get"]} PolicyRule{Resources:["globalconfigs"], APIGroups:["projectcalico.org"], Verbs:["list"]} PolicyRule{Resources:["globalconfigs"], APIGroups:["projectcalico.org"], Verbs:["update"]} PolicyRule{Resources:["globalconfigs"], APIGroups:["projectcalico.org"], Verbs:["watch"]} PolicyRule{Resources:["globalbgpconfigs"], APIGroups:["projectcalico.org"], Verbs:["create"]} PolicyRule{Resources:["globalbgpconfigs"], APIGroups:["projectcalico.org"], Verbs:["get"]} PolicyRule{Resources:["globalbgpconfigs"], APIGroups:["projectcalico.org"], Verbs:["list"]} PolicyRule{Resources:["globalbgpconfigs"], APIGroups:["projectcalico.org"], Verbs:["update"]} PolicyRule{Resources:["globalbgpconfigs"], APIGroups:["projectcalico.org"], Verbs:["watch"]} PolicyRule{Resources:["ippools"], APIGroups:["projectcalico.org"], Verbs:["create"]} PolicyRule{Resources:["ippools"], APIGroups:["projectcalico.org"], Verbs:["get"]} PolicyRule{Resources:["ippools"], APIGroups:["projectcalico.org"], Verbs:["list"]} PolicyRule{Resources:["ippools"], APIGroups:["projectcalico.org"], Verbs:["update"]} PolicyRule{Resources:["ippools"], APIGroups:["projectcalico.org"], Verbs:["watch"]} PolicyRule{Resources:["systemnetworkpolicies"], APIGroups:["alpha.projectcalico.org"], Verbs:["get"]} PolicyRule{Resources:["systemnetworkpolicies"], APIGroups:["alpha.projectcalico.org"], Verbs:["list"]} PolicyRule{Resources:["systemnetworkpolicies"], APIGroups:["alpha.projectcalico.org"], Verbs:["watch"]}] user=&{client  [system:authenticated] map[]} ownerrules=[] ruleResolutionErrors=[]
Error from server (Forbidden): error when creating "calico-daemonset.yaml": clusterroles.rbac.authorization.k8s.io "flannel" is forbidden: attempt to grant extra privileges: [PolicyRule{Resources:["pods"], APIGroups:[""], Verbs:["get"]} PolicyRule{Resources:["nodes"], APIGroups:[""], Verbs:["list"]} PolicyRule{Resources:["nodes"], APIGroups:[""], Verbs:["watch"]} PolicyRule{Resources:["nodes/status"], APIGroups:[""], Verbs:["patch"]}] user=&{client  [system:authenticated] map[]} ownerrules=[] ruleResolutionErrors=[]
```

**Comment by lachie83** [comment]
Resolved previously mentioned issue.

**Pull request merged**